src/main/content/_assets/js/guide.js

```diff
@@ -9,6 +9,89 @@
  *     IBM Corporation - initial API and implementation
  *******************************************************************************/
 
+// The background is shortened by 200px
+var backgroundSizeAdjustment = 200;
+
+function heightOfVisibleBackground() {
+    var result;
+    if(isBackgroundBottomVisible()) {
+        var scrollTop = $(window).scrollTop();
+        result = getBackgroundBottomPosition() - scrollTop;
+    } else {
+        // Assume the background is filling up the entire viewport
+        result = $(window).height();
+    }
+    return result;
+}
+
+function getBackgroundBottomPosition() {
+    var background = $('#background_container'),
+    elementTop = background.offset().top,
+    elementBottomPosition = elementTop + (background.outerHeight() - backgroundSizeAdjustment);
+    return elementBottomPosition;
+}
+
+// Determine if the bottom of the visible dark background is now visible 
+// in the browser's viewport.
+function isBackgroundBottomVisible() {
+    var background = $('#background_container'),
+        currentTopPosition = $(window).scrollTop(),
+        currentBottomPosition = currentTopPosition + $(window).height(),
+        elementBottomPosition = getBackgroundBottomPosition(),
+        visibleBottom = currentBottomPosition > elementBottomPosition;
+    return visibleBottom;
+}
+
+// Handle when to float the table of content
+function handleFloatingTableOfContent() {
+    if($(window).width() > 769) { // bootstrap grid size for sm column
+        // CURRENTLY IN DESKTOP VIEW
+        if($(window).scrollTop() > $('#toc_column').offset().top) {
+            // The top of the TOC is scrolling off the screen, enable floating TOC.
+
+            // Get the initial width of the TOC before applying position:fixed
+            var toc_width = $('#toc_inner').width();
+            
+            // position:fixed loses the original TOC width.  Restore original width when
+            // applying position fixed.
+            $('#toc_inner').width(toc_width);
+            if(isBackgroundBottomVisible()) {
+                handleTOCScolling();
+            } else {
+                // The entire viewport is filled with the background, so
+                // do not need to worry about the TOC flowing out of the background.
+                $('#toc_inner').css({"position":"fixed", "top":"0"});
```

Comment:
Would be more readable to have this as a function as well, similar to disableFloatingTOC();